scripts/nix-build.in

```diff
@@ -56,7 +56,7 @@ if ($runEnv && defined $ARGV[0] && $ARGV[0] !~ /nix-shell/) {
             @ARGV = ();
             while (<SCRIPT>) {
                 chomp;
-                if (/^\#\!\s*nix-shell (.*)$/) {
+                if (/^.*\s*nix-shell (.*)$/) {
```

Comment:
Erm, but then you need a space, don’t you?
That would be quite an arbitrary constraint imho.